framework/src/dgkernels/DGKernel.C

```diff
@@ -65,7 +65,8 @@ DGKernel::DGKernel(const InputParameters & parameters)
     FunctionInterface(this),
     UserObjectInterface(this),
     NeighborCoupleableMooseVariableDependencyIntermediateInterface(this, false, false),
-    TwoMaterialPropertyInterface(this),
+    TwoMaterialPropertyInterface(
+        this, boundaryRestricted() ? std::set<SubdomainID>() : blockIDs(), boundaryIDs()),
```

Comment:
At the very least you need to document the hack in the source code so we can find it later and remove it.